backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepository.java

```diff
@@ -222,56 +222,72 @@ public LiveOktaRepository(
     User user = users.single();
 
     String orgId = org.getExternalId();
-    boolean userInOrg = false;
-    // Remove user from old groups
-    for (Group g : user.listGroups()) {
-      if (g.getType() == GroupType.OKTA_GROUP
-          && g.getProfile().getName().startsWith(generateGroupOrgPrefix(orgId))) {
-        userInOrg = true;
-        // do not remove user from org's default group
-        if (!g.getProfile()
-            .getName()
-            .equals(generateRoleGroupName(orgId, OrganizationRole.getDefault()))) {
-          g.removeUser(user.getId());
-        }
-      }
-    }
-    if (!userInOrg) {
+
+    final String groupOrgPrefix = generateGroupOrgPrefix(orgId);
+    final String groupOrgDefaultName = generateRoleGroupName(orgId, OrganizationRole.getDefault());
+
+    Map<String, Group> currentOrgGroupMapForUser =
```

Comment:
nit: could you add a comment to explain what this String represents? At first glance, it's hard to find.